rm-main/src/ui/tabs/torrents/task_manager.rs

```diff
@@ -70,8 +100,25 @@ impl Component for TaskManager {
             CurrentTask::DeleteBar(delete_bar) => delete_bar.render(f, rect),
             CurrentTask::FilterBar(filter_bar) => filter_bar.render(f, rect),
             CurrentTask::Default(default_bar) => default_bar.render(f, rect),
+            CurrentTask::Status(status_bar) => status_bar.render(f, rect),
         }
     }
+
+    fn tick(&mut self) -> Option<Action> {
+        if let CurrentTask::Status(status_bar) = &mut self.current_task {
```

Comment:
Why not define a tick method for CurrentTask itself and handle things there?
self.current_task.tick() or self.current_task.handle_tick() so it's known that it's not from Component trait